lnrpc/routerrpc/forward_interceptor.go

```diff
@@ -125,52 +130,28 @@ func (r *forwardInterceptor) resolveFromClient(
 				)
 			}
 
-			// Verify that the size is equal to the fixed failure
-			// message size + hmac + two uint16 lengths. See BOLT
-			// #4.
-			if len(in.FailureMessage) !=
-				lnwire.FailureMessageLength+32+2+2 {
-
-				return status.Errorf(
-					codes.InvalidArgument,
-					"failure message length invalid",
-				)
+			fwdRes.FailureMessage = in.FailureMessage
+			fwdRes.FailureMessageEncrypted =
+				!in.FailureMessageUnencrypted
+		} else {
+			code, err := unmarshalFailCode(in.FailureCode)
+			if err != nil {
+				return err
 			}
 
-			return r.htlcSwitch.Resolve(&htlcswitch.FwdResolution{
-				Key:            circuitKey,
-				Action:         htlcswitch.FwdActionFail,
-				FailureMessage: in.FailureMessage,
-			})
+			fwdRes.FailureCode = code
 		}
 
-		var code lnwire.FailCode
-		switch in.FailureCode {
-		case lnrpc.Failure_INVALID_ONION_HMAC:
-			code = lnwire.CodeInvalidOnionHmac
-
-		case lnrpc.Failure_INVALID_ONION_KEY:
-			code = lnwire.CodeInvalidOnionKey
-
-		case lnrpc.Failure_INVALID_ONION_VERSION:
-			code = lnwire.CodeInvalidOnionVersion
+		err := r.htlcSwitch.Resolve(fwdRes)
+		if errors.Is(err, htlcswitch.ErrInvalidFailureMsgLength) ||
+			errors.Is(err, htlcswitch.ErrUnparseableFailureMsg) {
```

Comment:
Caused by `errorlint` where this could have been a clean and familiar switch statement. Urging reconsideration.

Reply:
Yes. Also trying to think of problems that we had in the past with not checking the whole chain. I don't think it really happened, or rarely.

In my opinion, it's not like for example binding the loop variable, which happened a lot and has serious consequences.

Reply:
what about 
```
switch {
case errors.Is(err, <err1>):
        fallthrough
case errors.Is(err, <err2>):
        return status.Errorf(...)
}
```

Reply:
Don't like it. I just don't think the `errors.Is` is helping us much. Almost like JavaScript, use == on any runtime determined type and it just works.